applications/FluidDynamicsApplication/custom_processes/weighted_divergence_calculation_process.h

```diff
@@ -0,0 +1,338 @@
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Riccardo Tosi
+
+#ifndef KRATOS_WEIGHTED_DIVERGENCE_CALCULATION_PROCESS_H
+#define KRATOS_WEIGHTED_DIVERGENCE_CALCULATION_PROCESS_H
+
+
+#include "includes/define.h"
+#include "includes/model_part.h"
+#include "includes/element.h"
+#include "includes/kratos_flags.h"
+#include "processes/process.h"
+#include "geometries/geometry.h"
+#include "utilities/geometry_utilities.h"
+#include "utilities/math_utils.h"
+#include "includes/kratos_parameters.h"
+#include "utilities/variable_utils.h"
+
+#include <string>
+#include <iostream>
+#include <sstream>
+
+#include <boost/functional/hash.hpp> //TODO: remove this dependence when Kratos has en internal one
+#include <unordered_map> //TODO: remove this dependence when Kratos has en internal one
+#include <utility>
+
+namespace Kratos
+{
+///@addtogroup FluidDynamicsApplication
+///@{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/// This process computes the element average in time of the divergence and of the seminorm of the velocity field.
+/// We define the seminorm of the velocity field as: \left \| \nabla u_{h} \right \|_{L^2(K)}^2 ,
+/// and the divergence as \left \| \nabla \cdot u_{h} \right \|_{L^2(K)}^2 ,
+/// where u is the velocity field and K an element of the domain \Omega.
+/// The time average does not consider the transient 20% first part of the simulation.
+/// The process requires a model part as input.
+
+/** Detail class definition.
+ */
+
+class WeightedDivergenceCalculationProcess: public Process
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of Process
+    KRATOS_CLASS_POINTER_DEFINITION(WeightedDivergenceCalculationProcess);
+
+    typedef ModelPart::ElementType ElementType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Constructor for WeightedDivergenceCalculationProcess Process
+    WeightedDivergenceCalculationProcess(ModelPart& rModelPart
+                    ):
+        Process(),
+        mrModelPart(rModelPart)
+    {
+    }
+
+    /// Destructor.
+    ~WeightedDivergenceCalculationProcess() override {}
+
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    /// This operator is provided to call the process as a function and simply calls the Execute method.
+    void operator()()
+    {
+        Execute();
+    }
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /// Execution of the process
+    void Execute() override
+    {
+        KRATOS_TRY;
+
+        // Set time coefficient: computations will be performed ONLY AFTER (time_coefficient * END_TIME)
+        const double time_coefficient = 0.2;
+
+        // Extract time information
+        const ProcessInfo& rCurrentProcessInfo = mrModelPart.GetProcessInfo();
```

Comment:
```suggestion
        const auto& r_current_processinfo = mrModelPart.GetProcessInfo();
```
(convention aims to use snake_case for local variables. Camel case is reserved for member variables and function arguments)

Reply:
Done the change for all the variables. A question: why is it better to use `auto&` than `ProcessInfo&`?

Reply:
If we eventually change the name of the current `ProcessInfo` class the `auto` will automatically deduce the new type, avoiding to change all the occurrencies in the code. Besides is easier while programming.